tuning/trainercontroller/controllermetrics/metrics.py

```diff
@@ -0,0 +1,218 @@
+import numpy as np
+from transformers import TrainerState, IntervalStrategy
+from transformers.utils import logging
+from collections import deque
+import math
+import copy
+
+logger = logging.get_logger(__name__)
+
+class MetricHandler:
+    """Base class for the controller-metrics"""
+    def validate(self, training_args) -> bool:
+        pass
+    
+    def compute(self, training_state, training_args=None, metrics=None) -> dict:
+        pass   
+
+class WindowStepLoss(MetricHandler):
+    """Implements the controller metric which evaluates loss-per-step over a user-defined window"""
+    
+    def __init__(self, window_size):
+        # Initialize the handler arguments
```

Comment:
I thought we discussed to have a cache in here too similar to epoch loss class, isn't it?

Reply:
addressed